rir/src/compiler/pir/pir_impl.h

```diff
@@ -0,0 +1,15 @@
+#ifndef PIR_PIR_IMPL_H
+#define PIR_PIR_IMPL_H
+
+// PIR includes. Please only include in .cpp!
```

Comment:
not sure if I get the question.

This is just the "all pir stuff" header file. So you should not include in other headers, since that gives cyclic dependency.

Why is type not there, probably since other included headers depend on type. but it could be added.